infrastructure/src/main/java/org/corfudb/infrastructure/CompactorService.java

```diff
@@ -82,7 +82,9 @@ public Runnable getSystemDownHandlerForCompactor(CorfuRuntime runtime) {
             //the following if condition can possibly invoked by all calls. It's ok to have multiple
             //calls to shutdown and start.
             synchronized (this) {
-                if (!corfuRuntimeOptional.isPresent() || runtime == corfuRuntimeOptional.get()) {
+                if (!corfuRuntimeOptional.isPresent()) {
```

Comment:
In the case of corfuRuntimeOptional is present and runtime != corfuRuntimeOptional, shouldn't we still shutdown runtime?

Reply:
You're right - I've addressed the missing case.